### ISSUE TITLE
GradingPrimaryTransform shader fails Windows / macOS

### ISSUE THREAD

**Issue by sobotka** [AcademySoftwareFoundation/OpenColorIO#1603]
`GradingPrimaryTransform` appears to fail in the shader on macOS (Intel). v 2.0.0 and 2.1.1.

```
ERROR (gpu.shader): OCIOShader FragShader: 
      | 
    1 | #version 330
      |                                                                                    ^
      | Error: Use of undeclared identifier 'inf'
      | 
      | #version 330
      |                                                                                     ^
      | Error: Use of undeclared identifier 'inf'
      | 
      | #version 330
      |                                                                                                  ^
      | Error: Use of undeclared identifier 'clampBlack'
      | Error: Use of undeclared identifier 'clampWhite'
```

**Comment by hodoulp** [comment]
In OCIOw e always have many [unit tests](https://github.com/AcademySoftwareFoundation/OpenColorIO/blob/main/tests/gpu/GradingPrimaryOp_test.cpp) around `GradingPrimaryTransform`, could you provide us some code snippet or a simplified config file to reproduce the problem?

**Comment by jessey-git** [comment]
@hodoulp 
I'm attempting to triage this issue from the originating software so here's some additional information:
- Library version: 2.0.0
- OCIO config: [config.ocio.txt](https://github.com/AcademySoftwareFoundation/OpenColorIO/files/8257104/config.ocio.txt)
- Shader dump
```
vec4 OCIO_to_display(in vec4 inPixel)
{
  vec4 outColor = inPixel;

  // Add GradingPrimary 'log' forward processing

  {
    vec3 brightness = vec3(0., 0., 0.);
    vec3 contrast = vec3(1.29999995, 1.29999995, 1.29999995);
    vec3 gamma = vec3(1., 1., 1.);
    float pivot = 0.400000006;
    float pivotBlack = 0.;
    float pivotWhite = 1.;
    float clampBlack = -inf;
    float clampWhite = inf;
    float saturation = 3.29999995;
    outColor.rgb += brightness;
    outColor.rgb = ( outColor.rgb - pivot ) * contrast + pivot;
    if ( gamma != vec3(1., 1., 1.) )
    {
      vec3 normalizedOut = abs(outColor.rgb - pivotBlack) / (pivotWhite - pivotBlack);
      vec3 scale = sign(outColor.rgb - pivotBlack) * (pivotWhite - pivotBlack);
      outColor.rgb = pow( normalizedOut, gamma ) * scale + pivotBlack;
    }
    vec3 lumaWgts = vec3(0.212599993, 0.715200007, 0.0722000003);
    float luma = dot( outColor.rgb, lumaWgts );
    outColor.rgb = luma + saturation * (outColor.rgb - luma);
    outColor.rgb = clamp( outColor.rgb, clampBlack, clampWhite );
  }

  return outColor;
}
```

It appears as though the 'clampWhite' and 'clampBlack' defaults of +,- max double is being written out to glsl as 'inf'

**Comment by jessey-git** [comment]
BTW, forgot to add that my repro was done on Windows+nVidia so it's not specific to Mac.  The only workaround is to specify your own values for clamp inside the config. For example: `clamp: {black: -100000, white: 100000}`

**Comment by hodoulp** [comment]
Hi @sobotka & @jessey-git 
The problem is fixed by PR #1462 (I'm able to reproduce the problem by removing a change from this PR) which is merged in `main` and `RB-2.1`. So, I suggest an update to OpenColorIO 2.1.

**Comment by doug-walker** [comment]
Despite what is written in the issue description above, based on Patrick's investigation, it should not be possible for the stated problem to occur in the 2.1 release, or later.  Therefore, I am closing this issue as Fixed.
